### PR TITLE
adding event listener to close navbar dropdown

### PR DESCRIPTION
**Pull request by s2sharpit**
## Related Issue

- Information about the related issue

Closes: #459 

### Describe the changes you've made

Enhance the profile dropdown functionality to automatically close when the user scrolls or clicks anywhere within the active window, providing a seamless and intuitive user experience.

## Type of change

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update.
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)

|      Original       |      Updated       |
| :-----------------: | :----------------: |
| original screenshot | updated screenshot |

## Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/WebXDAO/.github/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: src/components/main-nav.tsx [hunk 1]
```diff
@@ -6,32 +6,63 @@ import { cn } from "@/lib/utils";
...
 interface MainNavProps {
   items?: NavItem[];
 }
 
 export function MainNav({ items }: MainNavProps) {
   const [toggle, setToggle] = useState<boolean>(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(
+    (e: any) => {
```
Comment (thehashmap) [inline src/components/main-nav.tsx:24]: Instead of 'any' can we use a more specific event type? like MouseEvent or TouchEvent?

**Review by thehashmap — COMMENTED** [review]
It works as mentioned.
> But the dropdown does not close if you click the navbar anywhere other than the X button. Was that intended? If yes, then it's ok.
Otherwise, you can update for that too.

File: src/components/main-nav.tsx (hunk 1 above)
Comment (s2sharpit) [reply src/components/main-nav.tsx]: Hey, I have done it. Please check.

**Comment by s2sharpit** [comment]
> It works as mentioned.
> 
> > But the dropdown does not close if you click the navbar anywhere other than the X button. Was that intended? If yes, then it's ok.
> > Otherwise, you can update for that too.

Hey, It was not intentional. I have fixed it.

**Review by Anmol-Baranwal — COMMENTED** [review]
It works as mentioned, but there might be a couple of things that you can improve.

1. In the `handleClose` function, the event parameter is declared as `e: Event`. It would be more specific and accurate to specify the actual event type we are expecting.
e.g.
```typescript
(e: MouseEvent) => {
```

2. In the `useEffect` hook, the dependency array includes `handleClose` and toggle. It's generally recommended to only include variables or functions that are directly used inside the effect. Since `handleClose` is a callback function created with `useCallback`, its reference won't change unless its dependencies change. Therefore, you can omit it if you want from the dependency array to avoid unnecessary effect re-executions.
e.g.
```typescript
}, [toggle]);
```

@s2sharpit 
What do you think on this?

**Comment by s2sharpit** [comment]
Hey @Anmol-Baranwal 
I have optimized the code as per your suggestion. Please review it.

धन्यवाद 😊

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks clean to me.

**Comment by Anmol-Baranwal** [comment]
@s2sharpit 
Hey, I noticed you've been making a lot of contributions! Here's a suggestion to help you become an even better contributor in terms of writing commit messages: check out https://www.conventionalcommits.org/en/v1.0.0/. Following these guidelines can really boost your credibility. 

Also, it's best to use English instead of Hindi in your responses since not everyone may understand Hindi. Using a widely accessible language like English keeps things professional.

**Comment by s2sharpit** [comment]
> @s2sharpit Hey, I noticed you've been making a lot of contributions! Here's a suggestion to help you become an even better contributor in terms of writing commit messages: check out https://www.conventionalcommits.org/en/v1.0.0/. Following these guidelines can really boost your credibility.
> 
> Also, it's best to use English instead of Hindi in your responses since not everyone may understand Hindi. Using a widely accessible language like English keeps things professional.

Thank you, @Anmol-Baranwal, for your feedback and suggestion! I really appreciate your effort in helping me improve as a contributor. I will definitely check out the Conventional Commits guidelines you shared.

**Review by vinzvinci — APPROVED** [review]
LGTM ✅

**Comment by s2sharpit** [comment]
Hi @vinzvinci @Anmol-Baranwal @thehashmap ,
The pull request has been approved. Could you please take a look and merge it if everything looks good? 
Thank you!

**Comment by Anmol-Baranwal** [comment]
> Hi @vinzvinci @Anmol-Baranwal @thehashmap , The pull request has been approved. Could you please take a look and merge it if everything looks good? Thank you!

We will soon get this done. Kindly wait.

**Review by thehashmap — APPROVED** [review]

**Pull request merged**